alchemy/src/resend/audience.ts

```diff
@@ -0,0 +1,100 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { createResend, type ResendProps } from "./api.ts";
+
+export interface AudienceProps extends ResendProps {
+  /**
+   * The name of the audience.
+   * @default ${app}-${stage}-${id}
+   */
+  name?: string;
+}
+
+export type Audience = {
+  /**
+   * The ID of the audience.
+   */
+  id: string;
+  /**
+   * The name of the audience.
+   */
+  name: string;
+  /**
+   * The date and time the audience was created.
+   */
+  createdAt: Date;
+};
+
+export async function Audience(
+  id: string,
+  props: AudienceProps = {},
+): Promise<Audience> {
+  return await _Audience(id, props);
+}
+
+export namespace Audience {
+  export interface GetProps extends ResendProps {
+    id: string;
+  }
+
+  export const get = async (props: GetProps): Promise<Audience> => {
+    const resend = createResend(props);
+    const { data } = await resend.getAudiencesById({
+      path: {
+        id: props.id,
+      },
+    });
+    return {
+      id: data.id!,
+      name: data.name!,
+      createdAt: new Date(data.created_at!),
+    };
+  };
+}
```

Comment:
This feels cleaner to me than a separate `AudienceRef` but happy to change to that to follow conventions.